pandera/pandera.py

```diff
@@ -208,6 +209,128 @@ def __call__(self, parent_schema, check_index, check_obj):
             return _vcheck(check_obj)
 
 
+class Hypothesis(Check):
+    """Extends Check to perform a hypothesis test on a Column, potentially
+    grouped by another column
+
+    Parameters
+    ----------
+    test : callable
+        A function to check a series schema.
+    relationship : str|callable
+        Represents what relationship conditions are imposed on the hypothesis
+        test. A function or lambda function can be supplied. If a string is
+        provided, a lambda function will be returned from
+        Hypothesis.relationships.
+        Available relationships are: "greater_than", "less_than", "not_equal"
+    groupby : str|list[str]|callable|None
+        If a string or list of strings is
+        provided, then these columns are used to group the Column Series by
+        `groupby`. If a callable is passed, the expected signature is
+        DataFrame -> DataFrameGroupby. The function has access to the
+        entire dataframe, but the Column.name is selected from
+        this DataFrameGroupby object so that a SeriesGroupBy object is
+        passed into `fn`.
+
+        Specifying this argument changes the `fn` signature to:
+
+        dict[str|tuple[str], Series] -> bool|pd.Series[bool]
+
+        Where specific groups can be obtained from the input dict.
+
+    groups : str|list[str]|None
+        The dict input to the `fn` callable will be constrained to the
+        groups specified by `groups`.
+    test_kwargs : dict
+        Key Word arguments to be supplied to the test.
+    relationship_kwargs : dict
+        Key Word arguments to be supplied to the relationship function.
+        e.g. `alpha` could be used to specify a threshold in a t-test.
+    """
+    def __init__(self, test, relationship, groupby=None, groups=None,
+                 test_kwargs=None, relationship_kwargs=None):
+        #self.test = partial(test, **test_kwargs)
+        self.test = partial(test, **{} if test_kwargs is None else test_kwargs)
+        self.relationship = partial(self.relationships(relationship),
+                                    **relationship_kwargs)
+        self.groupby = groupby
```

Comment:
I don't think `self.groupby` and `self.groups` have to be set at the Hypothesis level, since the `Check.__init__` will do that.